include/openmc/mgxs_interface.h

```diff
@@ -12,36 +12,71 @@
 namespace openmc {
 
 //==============================================================================
-// Global variables
+// Global MGXS data container structure
 //==============================================================================
 
-namespace data {
+struct MgxsInterface
+{
+  int num_energy_groups;
+  int num_delayed_groups;
 
-extern std::vector<Mgxs> nuclides_MG;
-extern std::vector<Mgxs> macro_xs;
-extern int num_energy_groups;
-extern int num_delayed_groups;
-extern std::vector<double> energy_bins;
-extern std::vector<double> energy_bin_avg;
-extern std::vector<double> rev_energy_bins;
+  // List of available names in the HDF5 file
+  std::vector<std::string> xs_names;
+  std::vector<std::string> xs_to_read;
+  std::vector<std::vector<double>> xs_temps_to_read;
 
-} // namespace data
+  // Name of the HDF5 file which contains mgxs
+  std::string cross_sections_path;
 
-//==============================================================================
-// Mgxs data loading interface methods
-//==============================================================================
+  std::vector<Mgxs> nuclides_MG;
+  std::vector<Mgxs> macro_xs;
+
+  std::vector<double> energy_bins;
+  std::vector<double> energy_bin_avg;
+  std::vector<double> rev_energy_bins;
+
+  // temperatues of each available nuclide
+  std::vector<std::vector<double>> nuc_temps;
+
+  MgxsInterface() = default;
 
-void read_mgxs();
+  // Construct from path to cross sections file, as well as a list
+  // of XS to read and the corresponding temperatures for each XS
+  MgxsInterface(const std::string& path_cross_sections,
+                const std::vector<std::string> xs_to_read,
+                const std::vector<std::vector<double>> xs_temps);
+  void set_nuclides_and_temperatures(std::vector<std::string> arg_xs_to_read,
+                                     std::vector<std::vector<double>> xs_temps);
 
-void
-add_mgxs(hid_t file_id, const std::string& name,
-     const std::vector<double>& temperature);
+  void init();
+
+  void add_mgxs(hid_t file_id, const std::string& name,
+         const std::vector<double>& temperature);
+
+  void create_macro_xs();
+
+  std::vector<std::vector<double>> get_mat_kTs();
+
+  // Reads just the header of the cross sections file, to find
+  // min & max energies as well as the available XS
+  void read_header(const std::string& path_cross_sections);
+};
+
+namespace data {
+  extern MgxsInterface mgInterface;
```

Comment:
Variable names should be lowercase with underscores (see [naming](https://docs.openmc.org/en/latest/devguide/styleguide.html#naming) conventions)

Reply:
I think for simplicity, perhaps we should just call this `mg`. That makes accessing things out of it relatively simple, e.g.
```C++
data::mg.num_energy_bins
```
What do you think @nelsonag?

Reply:
I like the idea a lot, and have already made the change. So, if you don't approve @nelsonag, I will happily remove.

Reply:
I agree with that naming